File: NextcloudTalk/User Interface/HeaderWithButton.m
/**
 * SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

#import "HeaderWithButton.h"

@interface HeaderWithButton ()

@property (strong, nonatomic) IBOutlet UIView *contentView;

@end

@implementation HeaderWithButton

- (instancetype)init
{
    self = [super init];
    
    if (self) {
        [[NSBundle mainBundle] loadNibNamed:@"HeaderWithButton" owner:self options:nil];

        _label.textColor = [UIColor secondaryLabelColor];
        
        [self addSubview:self.contentView];
        
        if ([UIView userInterfaceLayoutDirectionForSemanticContentAttribute:_label.semanticContentAttribute] == UIUserInterfaceLayoutDirectionRightToLeft) {
            _label.textAlignment = NSTextAlignmentRight;
            _button.contentHorizontalAlignment = UIControlContentHorizontalAlignmentLeft;
        } else {
            _label.textAlignment = NSTextAlignmentLeft;
            _button.contentHorizontalAlignment = UIControlContentHorizontalAlignmentRight;
        }
        
        self.contentView.frame = self.bounds;
    }
    
    return self;
}

@end

File: NextcloudTalk/User Interface/HeaderWithButton.swift
//
// SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
// SPDX-License-Identifier: GPL-3.0-or-later
//

import UIKit

@objcMembers class HeaderWithButton: UIView {

    @IBOutlet var contentView: UIView!
    @IBOutlet weak var label: UILabel!
    @IBOutlet weak var button: UIButton!

    override init(frame: CGRect) {
        super.init(frame: frame)
        commonInit()
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        commonInit()
    }

    private func commonInit() {
        Bundle.main.loadNibNamed("HeaderWithButton", owner: self, options: nil)

        self.label.textColor = .secondaryLabel

        self.addSubview(self.contentView)

        if UIView.userInterfaceLayoutDirection(for: self.label.semanticContentAttribute) == .rightToLeft {
            self.label.textAlignment = .right
            self.button.contentHorizontalAlignment = .left
        } else {
            self.label.textAlignment = .left
            self.button.contentHorizontalAlignment = .right
        }

        self.contentView.frame = self.bounds
    }
}
